db.go

```diff
@@ -127,48 +163,95 @@ func (x *dummyAuthenticator) ResetPasswordFinish(identity string, token string,
 	}
 }
 
-func (x *dummyAuthenticator) CreateIdentity(identity, password string) error {
+func (x *dummyUserStore) CreateIdentity(identity, password string) error {
 	x.usersLock.Lock()
 	defer x.usersLock.Unlock()
-	if _, exists := x.users[CanonicalizeIdentity(identity)]; !exists {
-		x.users[CanonicalizeIdentity(identity)] = &dummyUser{identity, password, ""}
+	user := getUser(x.users, identity)
+	if user == nil {
+		userId := generateUserId(x.users)
+		x.users[userId] = &dummyUser{userId, identity, password, ""}
 		return nil
 	} else {
 		return ErrIdentityExists
 	}
 }
 
-func (x *dummyAuthenticator) RenameIdentity(oldIdent, newIdent string) error {
+func (x *dummyUserStore) RenameIdentity(oldIdent, newIdent string) error {
 	x.usersLock.Lock()
 	defer x.usersLock.Unlock()
 
-	if _, exists := x.users[CanonicalizeIdentity(newIdent)]; exists {
-		return ErrIdentityExists
-	}
-
-	oldIdent = CanonicalizeIdentity(oldIdent)
 	newKey := CanonicalizeIdentity(newIdent)
+	newUser := getUser(x.users, newKey)
```

Comment:
Don't postpone `oldIdent = CanonicalizeIdentity(oldIdent)` until deeper into the function. This sort of thing is prone to producing bugs later one, perhaps with the 2nd or 3rd refactor. Instead, sanitize/canonicalize/etc your function arguments as soon as the function starts.